app/process-image.js

```diff
@@ -471,104 +541,56 @@ class ImageProcessing {
+				height: region.height
+			};
 
-				// basic level 0 processing by default
-				if (level === 0) {
-					new_image = new_image.scan(0, 0, region1.width, region1.height, this.filterNearBlackContent).blur(1);
-				} else {
-					new_image = new_image.scan(0, 0, region1.width, region1.height, this.filterNearBlackContent2).blur(1);
-				}
+			new Promise((resolve, reject) => {
+				let new_image = image.clone()
+					.crop(cropped_region.x, cropped_region.y, cropped_region.width, cropped_region.height)
+					.scale(2, Jimp.RESIZE_HERMITE);
 
-				new_image.getBuffer(Jimp.MIME_PNG, (err, image) => {
-					if (err) {
-						reject(err);
-					}
+				switch (level) {
+					case 0:
+						new_image = new_image.scan(0, 0, cropped_region.width * 2, cropped_region.height * 2, this.filterNearBlackContent);
+						break;
 
-					this.tesseract.recognize(image, this.time_tesseract_options)
-						.catch(err => reject(err))
-						.then(result => {
-							// basically strip out everything except spaces, colons, and battery % life, then match any typical time values
-							const match = this.tesseractProcessTime(result);
-							if (match && match.length) {
-								resolve({
-									image: new_image,
-									text: match[0],
-									result
-								});
-							} else {
-								resolve({
-									image: new_image,
-									result
-								});
-							}
-						});
-				});
-			}));
+					case 1:
+						new_image = new_image.scan(0, 0, cropped_region.width * 2, cropped_region.height * 2, this.filterNearWhiteContent);
+						break;
 
-			// this check looks at the top-right for time (android?)
-			promises.push(new Promise((resolve, reject) => {
-				let new_image = image.clone()
-					.crop(region2.x, region2.y, region2.width, region2.height);
+					case 2:
+						new_image = new_image.scan(0, 0, cropped_region.width * 2, cropped_region.height * 2, this.filterSemiBlackContent);
+						break;
 
-				// basic level 0 processing by default
-				if (level === 0) {
-					new_image = new_image.scan(0, 0, region1.width, region1.height, this.filterNearWhiteContent).blur(1);
-				} else {
-					new_image = new_image.scan(0, 0, region1.width, region1.height, this.filterNearWhiteContent2).blur(1);
+					case 3:
+						new_image = new_image.scan(0, 0, cropped_region.width * 2, cropped_region.height * 2, this.filterSemiWhiteContent);
+						break;
 				}
 
 				new_image.getBuffer(Jimp.MIME_PNG, (err, image) => {
 					if (err) {
 						reject(err);
 					}
 
-					this.tesseract.recognize(image, this.time_tesseract_options)
+					this.time_tesseract.recognize(image, this.time_tesseract_options)
 						.catch(err => reject(err))
 						.then(result => {
-							// basically strip out everything except spaces, colons, and battery % life, then match any typical time values
 							const match = this.tesseractProcessTime(result);
 							if (match && match.length) {
 								resolve({
 									image: new_image,
-									text: match[0],
+									text: match[1],
```

Comment:
This seems odd? why is the 2nd matched result being used instead of the 1st?

Reply:
match[0] means return the original string, match[1] is the first matching group in the regex